latch_cli/services/local_dev.py

```diff
@@ -0,0 +1,319 @@
+import asyncio
+import json
+import os
+import sys
+import termios
+import tty
+from pathlib import Path
+from typing import Union
+
+import aioconsole
+import asyncssh
+import boto3
+import websockets.client as websockets
+import websockets.exceptions
+from prompt_toolkit.auto_suggest import AutoSuggestFromHistory
+from prompt_toolkit.completion import NestedCompleter, PathCompleter
+from prompt_toolkit.history import FileHistory
+from prompt_toolkit.patch_stdout import patch_stdout
+from prompt_toolkit.shortcuts import PromptSession
+
+from latch_cli.config.latch import LatchConfig
+from latch_cli.tinyrequests import post
+from latch_cli.utils import (
+    TemporarySSHCredentials,
+    current_workspace,
+    retrieve_or_login,
+)
+
+config = LatchConfig()
+sdk_endpoints = config.sdk_endpoints
+
+QUIT_COMMANDS = ["quit", "exit"]
+EXEC_COMMANDS = ["shell", "run", "run-script"]
+LIST_COMMANDS = ["ls", "list-tasks"]
+
+COMMANDS = QUIT_COMMANDS + EXEC_COMMANDS + LIST_COMMANDS
+
+
+async def copy_files(ssh_conn: asyncssh.SSHClientConnection, pkg_root: Path):
+    # TODO(ayush) do something more sophisticated/only send over
+    # diffs or smth to make this more efficient (rsync?)
+
+    coroutines = []
+    for path in [
+        pkg_root.joinpath("wf"),
+        pkg_root.joinpath("data"),
+        pkg_root.joinpath("scripts"),
+    ]:
+        if path.exists():
+            coroutines.append(
+                asyncssh.scp(
+                    path,
+                    (ssh_conn, f"~/{pkg_root.name}"),
+                    recurse=True,
+                    block_size=16384,
```

Comment:
why this number?

Reply:
this is 16KiB which is actually a pretty common block size
I'd write
```suggestion
                    block_size=1024 * 16,
```
though